test/com/google/enterprise/adaptor/database/DatabaseAdaptorTest.java

```diff
@@ -2643,6 +2682,124 @@ public void testMetadataColumns_columnAlias() throws Exception {
     assertEquals(expected, response.getMetadata());
   }
 
+  @Test
+  public void testExtraMetadata_docAndExtra() throws Exception {
+    executeUpdate("create table data(id int, name varchar(200))");
+    executeUpdate("create table metadata(id int, other varchar(200))");
+    executeUpdate("insert into data(id, name) values (1001, 'Doe, John')");
+    executeUpdate("insert into metadata(id, other) values (1001, 'John 1')");
+    executeUpdate("insert into metadata(id, other) values (1001, 'John 2')");
+
+    Map<String, String> configEntries = new HashMap<String, String>();
+    configEntries.put("db.uniqueKey", "id:int");
+    configEntries.put("db.everyDocIdSql", "select * from data");
```

Comment:
Unused, here and the other new tests, so put it after the usual comment to that effect.